### ISSUE TITLE
Empty Diff form shows up on revisions grid double click

### ISSUE THREAD

**Issue by jbialobr** [gitextensions/gitextensions#4579]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

Empty Diff form shows up on revisions grid double click.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

1. Open a repo.
2. Double click on any revision.

**What is the expected behavior?**

Form should be populated with data.

**Environment you encounter the issue:**

- GitExtensions version: 4de80e2b68da4ca5fd386391f27ccd4928396cf4

**Did this work in previous version of GitExtensions (which)?**
Yes: 2.51

**Comment by mdonatas** [comment]
@gerhardol could this be because of your changes in 82b846809389a37c3c6840fb5fd84f5f1e3e5594?

**Comment by drewnoakes** [comment]
I attempt to fix this in #4582.

Didn't see this issue until after filing that PR. Will update the description there to link it to this issue.

@mdonatas I ran a bisect to find the commit that introduced the bug:

```text
ab8d864aadc97e51d4de3d2a2cf585d969ec9c6e is the first bad commit
commit ab8d864aadc97e51d4de3d2a2cf585d969ec9c6e
Author: Gerhard Olsson <gerhard.nospam@gmail.com>
Date:   Sun Mar 4 13:24:17 2018 +0100

    FileStatusList handling of parents and selected revision

    Resolves #4561
    Refactoring to consistently use Revision and Parents in FileStatusList
    FileStatusList forms should use that instead of assuming parents from RevisionGrid.GetSelectedRevisions().
    The changes will remove code too.

    Note: This commits enables corrections for #4387 and #4396 etc too.

    Review comments: typecasting for GitRevision, consistently use GitItemsWithParents

:040000 040000 39e9d177a45ec1386d4ae7a589bf846c5f06e2ea e6f2a5ae60f06e5ef1d9b7c9e8ce95e6ec3ae9bd M  GitUI
```
